src/sagemaker/jumpstart/cache.py

```diff
@@ -442,12 +433,23 @@ def _retrieval_function(
             formatted_body, _ = self._get_json_file(id_info, data_type)
             model_specs = JumpStartModelSpecs(formatted_body)
             utils.emit_logs_based_on_model_specs(model_specs, self.get_region(), self._s3_client)
-            return JumpStartCachedContentValue(formatted_content=model_specs)
+            return JumpStartCachedContentValue(
+                formatted_content=model_specs
+            )
 
-        if data_type == HubContentType.MODEL:
-            hub_name, _, model_name, model_version = hub_utils.get_info_from_hub_resource_arn(
-                id_info
+        if data_type == HubContentType.NOTEBOOK:
+            hub_name, _, notebook_name, notebook_version = hub_utils.get_info_from_hub_resource_arn(id_info)
+            response: Dict[str, Any] = self._sagemaker_session.describe_hub_content(
+                hub_name=hub_name,
+                hub_content_name=notebook_name,
+                hub_content_version=notebook_version,
+                hub_content_type=data_type,
             )
+            hub_notebook_description = DescribeHubContentResponse(response)
+            return JumpStartCachedContentValue(formatted_content=hub_notebook_description)
+
+        if data_type == HubContentType.MODEL:
+            hub_name, _, model_name, model_version = hub_utils.get_info_from_hub_resource_arn(id_info)
```

Comment:
i mentioned in a previous pr, but it's not good practice imo to have a function return a tuple of length >2 for which particular items correspond to certain things. better to have a data class in this case

Reply:
Yeah this is a tricky one since we are dealing with circular dependencies. We cannot really use `HubArnExtractedInfo` here cause of that.